libdirections-session/src/main/java/com/mapbox/navigation/directions/session/MapboxDirectionsSession.kt

```diff
@@ -4,12 +4,25 @@ import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
 import com.mapbox.geojson.Point
 import com.mapbox.navigation.base.route.DirectionsSession
-import com.mapbox.navigation.base.route.Route
 import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.model.Route
 import java.util.concurrent.CopyOnWriteArrayList
 
 @MapboxNavigationModule(MapboxNavigationModuleType.DirectionsSession, skipConfiguration = true)
-class MapboxDirectionsSession(private val router: Router) : DirectionsSession {
+class MapboxDirectionsSession(
+    private val router: Router,
+    private var origin: Point,
+    private var waypoints: List<Point>,
+    private var destination: Point,
+    routeObserver: DirectionsSession.RouteObserver?
```

Comment:
Can we remove this argument? The behavior is going to be duplicate what's already possible `registerRouteObserver`.

Reply:
yep, but we cannot guaranty that user receive any update, if request starts just after instance create

Reply:
When you register a listener, the update is immediately provided, https://github.com/mapbox/mapbox-navigation-android/pull/2240/files#diff-c72c6e43f39a2af510904cc65b652104L45, making this argument unnecessary.

Reply:
yes, but if response failed -> no updates. And even more, if this's first response user first receive `null`, and after this real response. really strange behaviour, don't like it
IMO need to rewrite this class

Reply:
Why do you find it strange? That's typically how the observer pattern works, even in the `LiveData`. In our case `null` is a perfectly valid state, it describes that there is no route, and we should deliver that event when an observer is registered.

Reply:
Personally for me `onFailure` looks like it's enough to notify user, that `route fetching` failed. @Guardiola31337 @Lebedinsky wdyt?

Reply:
yes, but we don't deliver anything in case if first response fail

Reply:
@RingerJK capturing from the chat today, check out this commit and feel free to cherry-pick it to this PR - https://github.com/mapbox/mapbox-navigation-android/commit/7ccbcb3a6e7020738b773e80eedb811f54ff4ca5.

We're dropping the support for a list of observers, this way we can achieve a completely clear behavior, without the edge cases you're describing :)